sstables/mx/reader.cc

```diff
@@ -1700,7 +1718,9 @@ class mx_sstable_mutation_reader : public mp_row_consumer_reader_mx {
             close_index_reader = _index_reader->close().finally([_ = std::move(_index_reader)] {});
         }
 
-        return when_all_succeed(std::move(close_context), std::move(close_index_reader)).discard_result().handle_exception([] (std::exception_ptr ep) {
+        auto sst = std::move(*this).detach_sstable();
```

Comment:
Above we manipulate _context directly, so here we could manipulate _sst directly too. detach_sstable() is not a public API, so &&-qualification seems unnecessary, and it's weird that only _sst is manipulated this way and _context is not.

Reply:
Makes sense. I overcomplicated it. Better to just move sst into continuation.